src/uu/pwd/src/pwd.rs

```diff
@@ -43,24 +130,48 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     let usage = usage();
 
     let matches = uu_app().usage(&usage[..]).get_matches_from(args);
+    let cwd = if matches.is_present(OPT_LOGICAL) {
+        logical_path()
+    } else {
+        physical_path()
+    }
+    .map_err_context(|| "failed to get current directory".to_owned())?;
 
-    match env::current_dir() {
-        Ok(logical_path) => {
-            if matches.is_present(OPT_LOGICAL) {
-                println!("{}", logical_path.display());
-            } else {
-                let physical_path = absolute_path(&logical_path)
-                    .map_err_context(|| "failed to get absolute path".to_string())?;
-                println!("{}", physical_path.display());
-            }
-        }
-        Err(e) => {
-            return Err(USimpleError::new(
-                1,
-                format!("failed to get current directory {}", e),
-            ))
-        }
-    };
+    // \\?\ is a prefix Windows gives to paths under certain circumstances,
+    // including when canonicalizing them.
+    // With the right extension trait we can remove it non-lossily, but
+    // we print it lossily anyway, so no reason to bother.
+    #[cfg(windows)]
+    let cwd = cwd
+        .to_string_lossy()
+        .strip_prefix(r"\\?\")
+        .map(Into::into)
+        .unwrap_or(cwd);
+
+    print_path(&cwd).map_err_context(|| "failed to print current directory".to_owned())?;
+
+    Ok(())
+}
+
+fn print_path(path: &Path) -> io::Result<()> {
+    let stdout = io::stdout();
+    let mut stdout = stdout.lock();
+
+    // On Unix we print non-lossily.
```

Comment:
Interesting... This sounds like the right thing to do, but I don't think other utilities handle `unix` differently (i.e. we always print `path.display()`)

Reply:
I felt it was worth it here because you're likely to actually capture and use the output programmatically, which you wouldn't do for e.g. error messages. I can't offhand think of that many other programs where printing filenames is the whole point (`basename` and friends, `ls`, `find`?).

I've mostly been using `path.display()` but it's worse than what GNU does:
```
$ touch $'\xff\n\\'
$ rmdir $'\xff\n\\'
rmdir: failed to remove ''$'\377\n''\': Not a directory
```
It would be nice to have something like that. Maybe not that exact format, printing the byte in octal is a weird choice, but right now we lose information without even sanitizing junk like newlines.